### PR TITLE
add logging

### PR DESCRIPTION
**Pull request by anzioka**
PR:
Log error when PartitionManager.getDefaultPartitions(organizationId) returns null. 

This is an attempt to figure out why CollectionsManager.addTypeToEntityTypeCollectionTemplate(entityTypeCollectionId, collectionTemplateType) results in an error: `null cannot be cast to non-null type com.geekbeast.rhizome.hazelcast.DelegatedIntList`

### CODE REVIEW

File: conductor-client/src/main/kotlin/com/openlattice/data/storage/partitions/PartitionManager.kt [hunk 1]
```diff
@@ -58,7 +58,12 @@ class PartitionManager @JvmOverloads constructor(
     }
 
     fun getDefaultPartitions(organizationId: UUID): List<Int> {
-        return organizations.executeOnKey(organizationId, OrganizationReadEntryProcessor { DelegatedIntList(it.partitions) }) as DelegatedIntList
+        val partitions = organizations.executeOnKey(organizationId, OrganizationReadEntryProcessor { DelegatedIntList(it.partitions) })
+        if (partitions == null) {
+            logger.error("ERROR: loading default partitions for organization {} returned null", organizationId);
```
Comment (UnsungHero97) [inline conductor-client/src/main/kotlin/com/openlattice/data/storage/partitions/PartitionManager.kt:63]: `logger.error` will already print the `ERROR` prefix.

```suggestion
            logger.error("loading default partitions for organization {} returned null", organizationId);
```

**Review by UnsungHero97 — APPROVED** [review]

**Review by UnsungHero97 — APPROVED** [review]

File: conductor-client/src/main/kotlin/com/openlattice/data/storage/partitions/PartitionManager.kt [hunk 2]
```diff
@@ -58,7 +58,10 @@ class PartitionManager @JvmOverloads constructor(
     }
 
     fun getDefaultPartitions(organizationId: UUID): List<Int> {
-        return organizations.executeOnKey(organizationId, OrganizationReadEntryProcessor { DelegatedIntList(it.partitions) }) as DelegatedIntList
+        val partitions = organizations.executeOnKey(organizationId, OrganizationReadEntryProcessor { DelegatedIntList(it.partitions) })
+        checkNotNull(partitions, {"default partitions for organization $organizationId should not be null"})
```
Comment (geekbeast) [inline conductor-client/src/main/kotlin/com/openlattice/data/storage/partitions/PartitionManager.kt:62]: Doesn't kotlin allow checkNotNull(...) {...}? Minor style change as template is only evaluated on null either way.

**Review by geekbeast — APPROVED** [review]
Style comment, lgtm

File: conductor-client/src/main/kotlin/com/openlattice/data/storage/partitions/PartitionManager.kt (hunk 2 above)
Comment (anzioka) [reply conductor-client/src/main/kotlin/com/openlattice/data/storage/partitions/PartitionManager.kt:62]: ah, yes
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/check-not-null.html

**Pull request merged**